### PR TITLE
Refactor full node constructor

### PR DESCRIPTION
**Pull request by Manav-Aggarwal**
## Overview

Closes: #1229 

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

**Review by MSevey — APPROVED** [review]
utack

File: node/full.go [hunk 1]
```diff
@@ -187,12 +167,80 @@ func newFullNode(
...
+	node.p2pClient.SetTxValidator(node.newTxValidator())
 
 	return node, nil
 }
 
+func initProxyApp(clientCreator proxy.ClientCreator, logger log.Logger) (proxy.AppConns, error) {
+	proxyApp := proxy.NewAppConns(clientCreator, proxy.NopMetrics())
+	proxyApp.SetLogger(logger.With("module", "proxy"))
+	if err := proxyApp.Start(); err != nil {
+		return nil, fmt.Errorf("error starting proxy app connections: %v", err)
```
Comment (tuxcanfly) [inline node/full.go:179]: Might be worth wrapp this error.

**Review by tuxcanfly — APPROVED** [review]
OK.

Minor nit but might be worth revisiting to have some consistency in the error messages here. For e.g. we have:

* error starting proxy app connections
* couldn't get data availability client
* data availability layer client initialization error
* HeaderExchangeService initialization error

all of which use slightly different cases / verbs.

File: node/full.go (hunk 1 above)
Comment (Manav-Aggarwal) [reply node/full.go:179]: Using `fmt.Errorf` is equivalent to wrapping.

**Pull request merged**